### PR TITLE
Process and add flattened node list to graph

### LINKED ISSUE #30: STORE_NODE reducer function
The `STORE_NODE` reducer function should 
* Ensure the `id` field is populated
* Assign blank node `id`s if necessary.
* Flatten results (any `{}`-type embedded nodes should be stored separately and linked via their ID or a blank node identifier.
* blank node identifier assignment should be deterministic.

Also: evaluate factors that would make it necessary for this to be a task, like if it needs to emit additional actions/add new tasks.

### PR DESCRIPTION
**Pull request by ottonomy**
Add and flatten nodes to the graph in application state. Closes #30

### CODE REVIEW

File: badgecheck/state.py [hunk 1]
```diff
@@ -5,5 +5,17 @@
...
 def filter_active_tasks(state):
     return [t for t in state.get('tasks') if not t.get('complete')]
+
+
+# Graph
+def get_node_by_id(state, node_id):
+    """
+    Filter state to return first node that matches the requested id.
+    :param state: state object with "graph" property as a list
+    :param node_id: IRI-format string
```
Comment (ViktorHaag) [inline badgecheck/state.py]: If caller is looking for a blank node, they'll have to pass node_id in the form `'_b:{}'.format(node_id)` then, correct? Or once a node has been blank-node-ized, we consider the entire IRI to be the (new) node ID?

File: badgecheck/reducers/graph.py [hunk 2]
```diff
@@ -1,16 +1,45 @@
+import copy
+
 from ..actions.action_types import ADD_NODE, UPDATE_NODE
 
+current_node_id = -1
```
Comment (ViktorHaag) [inline badgecheck/reducers/graph.py]: minor: maybe we want to name the counter something other than 'node_id' as we use the term elsewhere to refer to the node.id properties value, whereas this counter is only a part of that.

**Comment by ViktorHaag** [comment]
Other than my minor points, this looks OK to me.

File: badgecheck/state.py (hunk 1 above)
Comment (ottonomy) [inline badgecheck/state.py]: I think this is correct, though I agree with your comment on the variable name for `current_node_id` (which I'm changing to `current_blank_node_number`). An "id" should be an IRI string, and that's how they'll find it in the documents. A caller will never have just the id number, they'll be looking at some other document that will reference an embedded node by an IRI, which might be a HTTP IRI or might be a blank node identifier.

File: badgecheck/reducers/graph.py (hunk 2 above)
Comment (ottonomy) [inline badgecheck/reducers/graph.py]: Agree. Changing to `current_blank_node_number` to make it clear that this number is not an id itself, it's just a number that helps us assign blank ids as needed.

**Pull request merged**